src/openrct2/ride/RideRatings.cpp

```diff
@@ -739,32 +739,37 @@ static void ride_ratings_calculate(RideRatingUpdateState& state, Ride* ride)
 #endif
 
 #ifdef ENABLE_SCRIPTING
-    auto& hookEngine = GetContext()->GetScriptEngine().GetHookEngine();
-    if (hookEngine.HasSubscriptions(HOOK_TYPE::RIDE_RATINGS_CALCULATE))
+    // Only call the 'ride.ratings.calculate' API hook if testing of the ride is complete
+    // https://github.com/OpenRCT2/OpenRCT2/issues/17844
```

Comment:
I don't see any reason to link back to the issue you can just git blame and the line above tells you all you need to know.

I was thinking about this issue yesterday and really we should address the issue a different way but it would require too much refactoring so this will do for now.

Reply:
Yes, fair comment. I'll remove the link.

What did you have in mind in terms of refactoring? An alternative I have considered is adding a boolean flag named `isComplete` to the `RideRatingsCalculateArgs` interface on the API so the plugin can check for itself whether it wants to do anything. However, I think this would only be useful to plugin authors if they depend on the 'premature' hook calls for something; I can't think of a reason why they would.

Reply:
I was thinking it would make more sense to have the hook where the ratings are set which is already behind one of these checks but it's currently set in multiple functions.

Reply:
I see. Yes, I did notice that all each of the ride specific `ride_ratings_calculate_` methods had the same guard clause to prevent execution if the ride hasn't finished testing and wondered why this is duplicated. Though, some methods don't have this guard clause immediately at the top of the method, for example `ride_ratings_calculate_boat_hire` and many of the flat rides such as `ride_ratings_calculate_3d_cinema` and stalls such as `ride_ratings_calculate_toilets`

Reply:
Ratings are behind a whole host of tests so its unlikely that a regression would be missed. That bit of code at the end is unfortunately not the same for all rides. I too thought the same as you. BUT it could be made the same with a bit of rework. Eitherway its for a later day.

Reply:
So we're happy with this quick fix for now?

Reply:
Yes